### PR TITLE
chore: refine e2e test scripts

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

- Added `-j` to make commands.
- Added `--depth=1` to git clone.
- Fixed Shellcheck lint errors.

### CODE REVIEW

File: scripts/integration-tests/e2e-babel.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@
 #==============================================================================#
 
 # Start in scripts/integration-tests/ even if run from root directory
-cd "$(dirname "$0")"
+cd "$(dirname "$0")" || exit
```
Comment (JLHwung) [inline scripts/integration-tests/e2e-babel.sh]: https://github.com/koalaman/shellcheck/wiki/SC2164

File: scripts/integration-tests/e2e-create-react-app.sh [hunk 2]
```diff
@@ -25,7 +25,7 @@ startLocalRegistry "$PWD"/../../verdaccio-config.yml
 yarn install
 # "yarn upgrade --scope @babel --latest" doesn't seem to work.
 # a means "all", while \n is the enter needed to confirm the selection.
-echo "a\n" | yarn upgrade-interactive --scope @babel --latest
+printf "a\n" | yarn upgrade-interactive --scope @babel --latest
```
Comment (JLHwung) [inline scripts/integration-tests/e2e-create-react-app.sh]: https://github.com/koalaman/shellcheck/wiki/SC2028

**Comment by nicolo-ribaudo** [comment]
Can we run shellcheck while lintin?

**Comment by JLHwung** [comment]
@nicolo-ribaudo Sure. But `shellcheck` is not a JavaScript program (written in Haskell). If contributor wants to add `shellcheck`, they has to install it by themself.

I can add an optional `lint-sh` make recipe and run it on travis. And we can add a `shellcheck` section to the contribution docs, in case one comes across the `shellcheck` errors.

**Comment by nicolo-ribaudo** [comment]
Yeah making it optional sounds good. But if this PR is blocking the other one, let's do it in a separate PR.

**Pull request merged**